src/solver/mha/mha_solver_backward.cpp

```diff
@@ -85,52 +86,75 @@ bool MhaBackward::IsApplicable([[maybe_unused]] const ExecutionContext& context,
-
-    return !miopen::IsDisabled(ENV(MIOPEN_DEBUG_ATTN_NAIVE_BWD)) //
-           && S <= std::numeric_limits<uint32_t>::max()          //
-           && D <= std::numeric_limits<uint32_t>::max()          //
-           && descsBackward.kDesc.IsPacked()                     //
-           && descsBackward.qDesc.IsPacked()                     //
-           && descsBackward.vDesc.IsPacked()                     //
-           && descsBackward.oDesc.IsPacked()                     //
-           && descsBackward.doDesc.IsPacked()                    //
-           && descsBackward.mDesc.IsPacked()                     //
-           && descsBackward.zInvDesc.IsPacked()                  //
-           && descsBackward.dkDesc.IsPacked()                    //
-           && descsBackward.dqDesc.IsPacked()                    //
-           && descsBackward.dvDesc.IsPacked()                    //
-           && MIOPEN_USE_GEMM;
+    const miopen::mha::MhaInputDescsBackward& descsBwd = problem.GetDescsBackward();
+
+    auto [N, H, S, D] = miopen::tien<4>(descsBwd.kDesc.GetLengths());
+
+    return MIOPEN_USE_GEMM                                            //
+           && !miopen::IsDisabled(ENV(MIOPEN_DEBUG_ATTN_NAIVE_BWD))   //
+           && S <= std::numeric_limits<uint32_t>::max()               //
+           && D <= std::numeric_limits<uint32_t>::max()               //
+           && descsBwd.kDesc.IsPacked()                               //
+           && descsBwd.qDesc.IsPacked()                               //
+           && descsBwd.vDesc.IsPacked()                               //
+           && descsBwd.oDesc.IsPacked()                               //
+           && descsBwd.doDesc.IsPacked()                              //
+           && descsBwd.mDesc.IsPacked()                               //
+           && descsBwd.zInvDesc.IsPacked()                            //
+           && descsBwd.dkDesc.IsPacked()                              //
+           && descsBwd.dqDesc.IsPacked()                              //
+           && descsBwd.dvDesc.IsPacked()                              //
+           && descsBwd.mDesc.GetType() == miopenFloat                 //
+           && descsBwd.zInvDesc.GetType() == miopenFloat              //
+           && descsBwd.kDesc.GetType() == descsBwd.qDesc.GetType()    //
+           && descsBwd.kDesc.GetType() == descsBwd.vDesc.GetType()    //
+           && descsBwd.kDesc.GetType() == descsBwd.oDesc.GetType()    //
+           && descsBwd.kDesc.GetType() == descsBwd.dqDesc.GetType()   //
+           && descsBwd.kDesc.GetType() == descsBwd.dkDesc.GetType()   //
+           && descsBwd.kDesc.GetType() == descsBwd.dvDesc.GetType()   //
+           && ((descsBwd.kDesc.GetType() == miopenFloat)              //
+               || (USE_ROCBLAS_EX3                                    //
+                   && (MIOPEN_FP8_IEEE_EXPONENT_BIAS == 0)            //
```

Comment:
what is the significance of this check?

Reply:
We support this solver for fp32 types always, and for fp8 type if rocblas supports it.